collector/collector.go

```diff
@@ -3,161 +3,109 @@ package collector
+		metrics = append(metrics, m)
 	}
-	return 0, fmt.Errorf("metric \"%s\" not found", metric)
+	return metrics
 }
 
-func (c LCDCollector) buildRequest(ctx context.Context, contractAddress string, query interface{}) (*http.Request, error) {
-	url := fmt.Sprintf("%s/wasm/contracts/%s/store", c.LCDEndpoint, contractAddress)
-	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to prapare request for %+v query: %w", query, err)
-	}
-	q := req.URL.Query()
-	queryRaw, err := json.Marshal(query)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal %+v query: %w", query, err)
-	}
-	q.Add("query_msg", string(queryRaw))
-	req.URL.RawQuery = q.Encode()
-	return req, nil
+func (c *LCDCollector) SetTransport(transport runtime.ClientTransport) {
+	c.apiClient.SetTransport(transport)
 }
 
-func (c LCDCollector) processRequest(req *http.Request, ret interface{}) error {
-	resp, err := c.HttpClient.Do(req)
-	if err != nil {
-		return fmt.Errorf("failed to get request: %w", err)
-	}
-
-	err = parseRequestBody(resp, ret)
-	if err != nil {
-		return fmt.Errorf("failed to process response body: %w", err)
+func (c LCDCollector) Get(metric Metrics) (float64, error) {
+	monitor, found := c.Metrics[metric]
+	if !found {
+		return 0, fmt.Errorf("monitor for metric \"%s\" not found", metric)
 	}
-	return nil
+	return monitor.Get(metric)
 }
 
-func (c LCDCollector) buildAndProcessRequest(
-	ctx context.Context,
-	contractAddress string,
-	query interface{},
-	ret interface{},
-) error {
-	req, err := c.buildRequest(ctx, contractAddress, query)
-	if err != nil {
-		return fmt.Errorf(
-			"failed to build request %+v for %s contract: %w",
-			query,
-			contractAddress,
-			err,
-		)
-	}
-	err = c.processRequest(req, ret)
-	if err != nil {
-		return fmt.Errorf("failed to process %+v request for %s contract: %w", query, contractAddress, err)
+func (c *LCDCollector) UpdateData(ctx context.Context) error {
+	for _, monitor := range c.Monitors {
+		err := monitor.Handler(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to update data: %w", err)
+		}
 	}
 	return nil
 }
 
-func (c *LCDCollector) getRewardState(ctx context.Context) (*RewardStateResponse, error) {
-	req, resp := GetRewardResponseStatePair()
-	err := c.buildAndProcessRequest(ctx, c.RewardContractAddress, req, &resp)
-	if err != nil {
-		return nil, fmt.Errorf("failed to process RewardState request: %w", err)
-	}
-	c.logger.Infoln("updated reward state info")
-	return &resp, nil
-}
-
-func (c *LCDCollector) getBlunaTokenInfo(ctx context.Context) (*TokenInfoResponse, error) {
-	req, resp := GetCommonTokenInfoPair()
-	err := c.buildAndProcessRequest(ctx, c.BlunaContractAddress, req, &resp)
-	if err != nil {
-		return nil, fmt.Errorf("failed to process BlunaTokenInfo request: %w", err)
+func (c *LCDCollector) RegisterMonitor(m Monitor) {
+	for _, metric := range m.ProvidedMetrics() {
+		if founded, found := c.Metrics[metric]; found {
```

Comment:
founded is a grammar mistake, please rename

Reply:
renamed to wantedMonitor